[test] Autocomplete drop "defaultProps" pattern

eps1lon opened the pull request:
Removes indirection of having to look up what props are set (motivated by having to check out what `openOnFocus` is supposed to do in #20286). It's apparent from the current state of the test suite that these props aren't actually default props anymore.

eps1lon commented:
> Are you sure about pruning openOnFocus from all the tests? Maybe it was required somewhere to create a failing test case.

No tests were failing and are now. What do you mean by failing test case?

If `openOnFocus` is added in every test then this is indicative that we want it as the default behavior anyway.

oliviertassinari reviewed (approved):
I didn't look at each test case to make sure the new conditions are correct, I trust you for that

oliviertassinari commented:
Rebased on next hoping it will solve the codecov fail.

eps1lon commented:
> Consider the following. x weeks ago we had a bug. We added a failing test case. We fixed the bug. What if the failing test case was relying on openOnFocus?

Exactly my point why the defaultProps pattern is problematic: You don't know if the props are relevant to the test or not. I should've fixed this earlier but at least it's still during alpha.
